### PR TITLE
Gcp Function Middleware

### PR DESCRIPTION
**Pull request by girishd**
## Summary

Added a new GPT Action - Google Cloud Function. I will be following up with a Pinecone GPT soon

## Motivation

This should serve as the template to build other Google Cloud Function actions that connect to 3rd party systems - like SQL databases, etc

---

## For new content

When contributing new content, read through our [contribution guidelines](https://github.com/openai/openai-cookbook/blob/main/CONTRIBUTING.md), and mark the following action items as completed:

- [yes ] I have added a new entry in [registry.yaml](https://github.com/openai/openai-cookbook/blob/main/registry.yaml) (and, optionally, in [authors.yaml](https://github.com/openai/openai-cookbook/blob/main/authors.yaml)) so that my content renders on the cookbook website.
- [yes ] I have conducted a self-review of my content based on the [contribution guidelines](https://github.com/openai/openai-cookbook/blob/main/CONTRIBUTING.md#rubric):
  - [ yes] Relevance: This content is related to building with OpenAI technologies and is useful to others.
  - [yes ] Uniqueness: I have searched for related examples in the OpenAI Cookbook, and verified that my content offers new insights or unique information compared to existing documentation.
  - [yes ] Spelling and Grammar: I have checked for spelling or grammatical mistakes.
  - [ yes] Clarity: I have done a final read-through and verified that my submission is well-organized and easy to understand.
  - [yes ] Correctness: The information I include is correct and all of my code executes successfully.
  - [yes ] Completeness: I have explained everything fully, including all necessary references and citations.

We will rate each of these areas on a scale from 1 to 4, and will only accept contributions that score 3 or higher on all areas. Refer to our [contribution guidelines](https://github.com/openai/openai-cookbook/blob/main/CONTRIBUTING.md) for more details.

### CODE REVIEW

File: images/gcp-function-middleware-oauthcallback.png
Comment (aaronwilkowitz-openai) [inline images/gcp-function-middleware-oauthcallback.png:1]: Can you please black out the callback URL? It's possibly sensitive (And even if not, it's good practice to show our customers reading our docs.)

**Review by aaronwilkowitz-openai — COMMENTED** [review]
Looks good, @girishd ! I added one small comment on one of the photos to edit. Thanks!

**Comment by girishd** [comment]
Done!

**Review by aaronwilkowitz-openai — APPROVED** [review]
Looks great, thanks!

**Pull request merged**